### PR TITLE
Run `.buildbot.sh` through shellcheck.

### PR DESCRIPTION
**Pull request by ltratt**
Every time I do this, I learn something new about the shell...

### CODE REVIEW

File: .buildbot.sh [hunk 1]
```diff
@@ -29,10 +31,10 @@ cargo fmt --all -- --check
 cd ykllvm
 ykllvm_hash=$(git rev-parse HEAD)
 cached_ykllvm=0
-if [ -f /opt/ykllvm_cache/ykllvm-release-with-assertions-${ykllvm_hash}.tgz ]; then
+if [ -f /opt/ykllvm_cache/ykllvm-release-with-assertions-"${ykllvm_hash}".tgz ]; then
```
Comment (vext01) [inline .buildbot.sh:34]: Go on then. Why is this quoting better?

Comment (ltratt) [reply .buildbot.sh:34]: It stops word splitting. Which I didn't even realise!

**Comment by ltratt** [comment]
@vext01 I am not sure that test failure relates to this PR? Maybe worth kicking again to see.

**Comment by ltratt** [comment]
@vext01 This one should hopefully merge now.

**Pull request merged**